### ISSUE TITLE
Unexpected failure if signfile not exist: bad substitution

### ISSUE THREAD

**Issue by recolic** [dell/dkms#236]
## Background

At <https://github.com/dell/dkms/blob/1e24a54acaa2e3a8aa6e8e24b21f47553370c4fe/dkms.in#L864>, it's failing because signfile does not exist. This is a Arch Linux machine with modified /etc/os-release, to trick some other programs. 

## What did I do

```
dkms install --no-depmod nvidia/515.57 -k 5.18.11-hardened1-1-hardened
```

## Expected behavior

Getting a warning: `Binary sign-file not found, module won't be signed`, and then successfully install the dkms module. 

## Actual Behavior

Getting a fatal error: 

```
==> dkms install --no-depmod nvidia/515.57 -k 5.18.11-hardened1-1-hardened
/usr/bin/dkms: line 864: /usr/lib/linux-kbuild-${$kernelver%.*}/scripts/sign-file: bad substitution
==> WARNING: `dkms install --no-depmod nvidia/515.57 -k 5.18.11-hardened1-1-hardened' exited 1
```

## System Info

ArchLinux, bash 5.1.016-1, dkms-3.0.4, kernel 5.18.11-hardened1-1-hardened

**Comment by scaronni** [comment]
Fixed in #230.

**Comment by recolic** [comment]
Hi. This PR is not fixed by PR #230 .  It's a bash command substitution issue, and it's not related to Arch Linux support. 

I could still reproduce this issue in dkms 3.0.6.  

![image](https://user-images.githubusercontent.com/19610256/179168619-3dbde931-0203-46dd-8f0e-acdc68d4bc73.png)

**Comment by recolic** [comment]
I will suggest a PR to solve this issue soon.
